pkg/minishift/provisioner/minishift_detector_test.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright (C) 2016 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provisioner
+
+import (
+	"testing"
+
+	"github.com/docker/machine/libmachine/provision"
+)
+
+func TestMinishiftDetector(t *testing.T){
```

Comment:
Ok, this tests _isMinishiftIso_. Fair enough. Good test. It still does not test the actual provider selection. I think we could go a step further here. There is a lot which can fall through here.